In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [14]:
pd.set_option('display.max_columns', None)

In [3]:
df = pd.read_excel('../data/raw/raw_data.xlsx')

In [4]:
df.head()

,SKU_ID,Category,ABC_Class,Supplier_Name,Warehouse_Location,Received_Date,Last_Purchase_Date,Stock_Age_Days,Quantity_On_Hand,Damaged_Qty,Returns_Qty,Avg_Daily_Sales,Forecast_Next_30d,Reorder_Point,Safety_Stock,Lead_Time_Days,Total_Inventory_Value_USD,Supplier_OnTime_Pct,Inventory_Status
0,SKU_14996,Household,A,NextGen Traders,Chennai Warehouse,2024-06-11,2024-07-11,212,482.0,13,9,19.84,459.32,126,47,7.0,45555.63,82.64,Normal
1,SKU_10247,Snacks,C,NextGen Traders,Bangalore Warehouse,2024-11-24,2025-03-19,357,640.0,8,6,34.81,1072.54,110,75,5.0,36577.48,94.58,Normal
2,SKU_16215,Dairy,C,NextGen Traders,Chennai Warehouse,2023-02-01,2023-03-06,104,538.0,5,3,49.66,1891.81,118,66,12.0,23742.67,98.22,Overstock
3,SKU_13545,Bakery,A,Prime Distribution,Chennai Warehouse,2024-03-13,2024-06-04,161,530.0,9,2,20.00,662.42,119,11,8.0,49287.57,97.45,Low Stock
4,SKU_15397,Produce,B,Global Retail Supply,Chennai Warehouse,2023-05-01,2023-05-09,289,397.0,14,3,32.63,1115.41,128,42,3.0,16929.86,86.75,Overstock


### Observation
- Dataset contains inventory operational data.
- Includes supplier, warehouse, sales, and stock information.
- Date columns appear available for time-based analysis.

In [5]:
df.shape

(6712, 19)

In [6]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 6712 entries, 0 to 6711
Data columns (total 19 columns):
 #   Column                     Non-Null Count  Dtype         
---  ------                     --------------  -----         
 0   SKU_ID                     6712 non-null   str           
 1   Category                   6712 non-null   str           
 2   ABC_Class                  6712 non-null   str           
 3   Supplier_Name              6712 non-null   str           
 4   Warehouse_Location         6712 non-null   str           
 5   Received_Date              6511 non-null   datetime64[us]
 6   Last_Purchase_Date         6712 non-null   datetime64[us]
 7   Stock_Age_Days             6712 non-null   int64         
 8   Quantity_On_Hand           6712 non-null   float64       
 9   Damaged_Qty                6712 non-null   int64         
 10  Returns_Qty                6712 non-null   int64         
 11  Avg_Daily_Sales            6712 non-null   float64       
 12  Forecast_Next_30d

In [16]:
df.describe().T

,count,mean,min,25%,50%,75%,max,std
Received_Date,6511,2024-02-28 12:18:01.492858,2023-01-01 00:00:00,2023-07-28 00:00:00,2024-02-29 00:00:00,2024-09-30 00:00:00,2025-04-30 00:00:00,NaN
Last_Purchase_Date,6712,2024-04-29 20:31:02.216925,2023-01-08 00:00:00,2023-09-27 00:00:00,2024-04-27 00:00:00,2024-12-02 00:00:00,2025-08-25 00:00:00,NaN
Stock_Age_Days,6712.0,203.411651,5.0,103.0,205.0,302.0,400.0,114.147916
Quantity_On_Hand,6712.0,420.498784,5.0,339.0,421.0,500.0,1320.229621,122.066614
Damaged_Qty,6712.0,9.602354,0.0,6.0,10.0,13.0,28.0,4.865126
Returns_Qty,6712.0,7.546633,0.0,5.0,8.0,10.0,25.0,3.914893
Avg_Daily_Sales,6712.0,34.889443,1.0,28.08,35.0,41.75,70.29,10.028864
Forecast_Next_30d,6509.0,1044.936906,20.84,779.4,1017.63,1285.4,2478.37,366.038167
Reorder_Point,6712.0,119.042312,10.0,98.0,119.0,140.0,237.0,30.194679
Safety_Stock,6712.0,59.629917,5.0,46.0,60.0,73.0,127.0,19.980234


In [8]:
df.describe(include=['object', 'string'])

,SKU_ID,Category,ABC_Class,Supplier_Name,Warehouse_Location,Inventory_Status
count,6712,6712,6712,6712,6712,6712
unique,6700,8,3,6,5,4
top,SKU_15166,Produce,A,Prime Distribution,Kolkata Warehouse,Normal
freq,2,869,2296,1133,1372,1761


In [9]:
df.nunique().sort_values(ascending=False)

SKU_ID                       6700
Total_Inventory_Value_USD    6685
Forecast_Next_30d            6338
Avg_Daily_Sales              3249
Supplier_OnTime_Pct          1664
Last_Purchase_Date            941
Received_Date                 851
Quantity_On_Hand              635
Stock_Age_Days                396
Reorder_Point                 194
Safety_Stock                  120
Damaged_Qty                    28
Returns_Qty                    23
Lead_Time_Days                 21
Category                        8
Supplier_Name                   6
Warehouse_Location              5
Inventory_Status                4
ABC_Class                       3
dtype: int64

In [10]:
missing = df.isnull().sum()

missing_percent = (missing / len(df)) * 100

missing_df = pd.DataFrame({
    'Missing Values': missing,
    'Percentage': missing_percent
})

missing_df.sort_values(
    by='Percentage',
    ascending=False
)

,Missing Values,Percentage
Forecast_Next_30d,203,3.024434
Lead_Time_Days,202,3.009535
Supplier_OnTime_Pct,202,3.009535
Received_Date,201,2.994636
ABC_Class,0,0.000000
SKU_ID,0,0.000000
Category,0,0.000000
Stock_Age_Days,0,0.000000
Supplier_Name,0,0.000000
Warehouse_Location,0,0.000000


In [11]:
df.duplicated().sum()

np.int64(12)

In [13]:
num_cols = df.select_dtypes(include=np.number).columns.tolist()

cat_cols = df.select_dtypes(include=['object', 'string']).columns.tolist()

print("Numerical Columns:")
print(num_cols)

print("\nCategorical Columns:")
print(cat_cols)

Numerical Columns:
['Stock_Age_Days', 'Quantity_On_Hand', 'Damaged_Qty', 'Returns_Qty', 'Avg_Daily_Sales', 'Forecast_Next_30d', 'Reorder_Point', 'Safety_Stock', 'Lead_Time_Days', 'Total_Inventory_Value_USD', 'Supplier_OnTime_Pct']

Categorical Columns:
['SKU_ID', 'Category', 'ABC_Class', 'Supplier_Name', 'Warehouse_Location', 'Inventory_Status']


### Key Findings

- The dataset contains **6,712 inventory records** and **19 features**.
- The data includes a combination of **categorical, numerical, and date-related variables**.
- Missing values were identified in **Received_Date, Forecast_Next_30d, Lead_Time_Days, and Supplier_OnTime_Pct** need to be handled during the data cleaning .
- Duplicate records were detected and will need to be handled during the data cleaning stage.
- The dataset captures important inventory management information such as:
  - Product details
  - Stock levels
  - Sales performance
  - Demand forecasts
  - Reorder and safety stock levels
  - Supplier performance
  - Inventory value

### Business Relevance

The dataset can be used to analyze inventory health, understand demand patterns, evaluate supplier performance, identify stock risks, and support inventory planning decisions.